### ISSUE TITLE
mongorestore docker-compose network DNS failure

### ISSUE THREAD

**Issue by thebleucheese** [docker-library/mongo#273]
Setup:
docker-compose network with Node JS container, mongo container, and mongo-seed container.
The latter 2 use this base mongo image.

When I run "mongorestore --host mongo" (mongo being the docker container_name) from the mongo-seed container trying to connect to mongo on an effort to seed initial application data it fails with this error: Failed: error connecting to db server: no reachable servers.  When I use the IP address it works fine.  I can connect to the hostname using the mongo CLI command so it's not a DNS issue.

To further verify that this is a problem with this specific image, I ran apt-get install mongodb on my Node JS container and I was able to run mongorestore --host mongo without any issues.

To recap:
mongo container: 
mongo CLI tool - can connect to another mongo host with both IP and hostname
mongorestore command - can connect to another mongo host with IP but NOT hostname
mongoimport - same behavior as mongorestore

nodejs container: 
apt-get install mongodb
both mongorestore and mongo CLI tool can connect to another mongo host with both IP and hostname

My guess: some sort of dependency for mongorestore to resolve DNS has been removed or was not installed on the mongo image.

**Comment by yosifkit** [comment]
I am unable to reproduce:

```console
$ cat docker-compose.yml
version: '3.3'

services:
        mongo1:
          image: mongo
          networks:
                  - mongo
          environment:
                  - MONGO_INITDB_ROOT_USERNAME=root
                  - MONGO_INITDB_ROOT_PASSWORD=pass
        
        
        mongo2: 
          image: mongo
          networks:
                  - mongo
          environment:
                  - MONGO_INITDB_ROOT_USERNAME=root
                  - MONGO_INITDB_ROOT_PASSWORD=pass

networks:
  mongo:
$ docker-compose up -d
$ docker exec -it mongo-273_mongo2_1 bash
root@f5bbce808777:/# cd data/
root@f5bbce808777:/data# mkdir dump
root@f5bbce808777:/data# mongorestore --host mongo1
2018-05-18T20:17:49.360+0000	using default 'dump' directory
2018-05-18T20:17:49.360+0000	preparing collections to restore from
2018-05-18T20:17:49.360+0000	done
```

**Comment by wglambert** [comment]
Since this issue doesn't seem to denote any errors with the image I'm going to close, if there is something reproducible to show an issue with the image then I'll re-open.

**Comment by lerox** [comment]
Hi guys, I had a similar behaviour without passing explicitly the docker-compose version, and when I use the version 3 everything works well!

before:

```
# others

mongodb:
  image: mongo
  ports:
    - "27027"

```

after:
```
version: '3'

services:
	
  # others
 
  mongodb:
     image: mongo
     ports:
       - "27027"
```
